Source/RenderStream/Private/RenderStream.cpp

```diff
@@ -270,19 +272,41 @@ bool FRenderStreamModule::PopulateStreamPool()
 
         const RenderStreamLink::StreamDescriptions* header = nBytes >= sizeof(RenderStreamLink::StreamDescriptions) ? reinterpret_cast<const RenderStreamLink::StreamDescriptions*>(descMem.data()) : nullptr;
         const size_t numStreams = header ? header->nStreams : 0;
+        TArray<FStreamInfo> streamInfoArray;
         for (size_t i = 0; i < numStreams; ++i)
         {
             const RenderStreamLink::StreamDescription& description = header->streams[i];
             const FString Name(description.name);
             const FIntPoint Resolution(description.width, description.height);
             const FString Channel(description.channel);
-            if (!StreamPool->GetStream(Name))
+            streamInfoArray.Push({ Channel, Name });
+
+            if (TSharedPtr<FFrameStream> Stream = StreamPool->GetStream(Name))
             {
+                // Stream exists in pool but might have changed, update config
+                UE_LOG(LogRenderStream, Log, TEXT("Updating config for existing stream %s at %dx%d"), *Name, Resolution.X, Resolution.Y);
+                check(Resolution == Stream->Resolution());  // Changing stream resolution is not currently supported
+                Stream->Setup(Name, Resolution, Channel, description.clipping, description.handle, description.format);
```

Comment:
Setup calls RSUCHelpers::CreateStreamResources which creates a texture and fence. Texture looks to be refcounted, but fence looks to be a raw pointer.

Reply:
I'm not sure what the implications of this would be. This code isn't really needed for failover any more, as we don't update the config of existing streams, so I'll just remove it if it's likely to cause problems.